### PR TITLE
[ADD] Add Soundcloud to Files

### PR DESCRIPTION
**Pull request by flyingrub**

### CODE REVIEW

**Comment by flyingrub** [comment]
https://github.com/cozy/cozy-browser-sdk/pull/31

**Comment by flyingrub** [comment]
https://github.com/cozy/cozy-data-system/pull/253

File: app/libs/soundcloud.js [hunk 1]
```diff
@@ -24,83 +25,189 @@ class Soundcloud {
...
             }
         });
     }
 
+    // Create a promise for each track
+    launchImport(tracks, playlist) {
+        let promises = [];
+        for (let i = 0; i < tracks.length; i++) {
+            let track = tracks[i];
+            track.playlist = playlist;
```
Comment (aenario) [inline app/libs/soundcloud.js:39]: is cleaner to use `tracks.map` than an array and a loop.

File: app/libs/soundcloud.js [hunk 2]
```diff
@@ -24,83 +25,189 @@ class Soundcloud {
...
+            let promise = new Promise((resolve, reject) => {
```
Comment (aenario) [inline app/libs/soundcloud.js]: Using Promise constructor is almost always an anti-pattern.
In this case, you can do something like

```
if(!track.streamable){
     promise = Promise.resolve(.... not streamable)
}else{
       promise = this.checkAlreadyExists(track)
}
```

You will need to change checkAlreadyExist to return a Promise.
The global idea, is that mixin Promise & callbacks make it weird, use them everywhere (from cozy-browser-sdk) or dont use them.

File: app/libs/soundcloud.js [hunk 3]
```diff
@@ -134,6 +241,42 @@ class Soundcloud {
...
+
+            for (let i = 0; i < values.length; i++) {
+                let el = values[i];
+                if (el.status == 'ko') {
+                    notification = {
+                        status: 'ko',
+                        message: t('one track not imported')
+                    }
+                }
+                break;
```
Comment (aenario) [inline app/libs/soundcloud.js]: Not sure what you are trying to do, but with the break here, this code is equivalent to 

```
if(values[0].status == 'ko') notification = {...}
```

Comment (flyingrub) [inline app/libs/soundcloud.js]: the break is used to break out of the for loop if as soon as an `el.status == 'ko'`.

**Comment by m4dz** [comment]
Is this one ready to merge ?

**Comment by flyingrub** [comment]
i think so :)

**Pull request merged**